app/src/main/java/pm/gnosis/heimdall/ui/onboarding/account/create/GenerateMnemonicActivity.kt

```diff
@@ -56,7 +57,12 @@ class GenerateMnemonicActivity : SecuredBaseActivity() {
             .subscribeForResult(onNext = ::onMnemonic, onError = ::onMnemonicError)
 
         disposables += confirmDialogClick
-            .flatMapSingle { viewModel.saveAccountWithMnemonic(it) }
+            .flatMapSingle {
+                viewModel.saveAccountWithMnemonic(it)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .doOnSubscribe { layout_generate_mnemonic_progress_bar.visibility = View.VISIBLE }
+                    .doOnEvent { _, _ -> layout_generate_mnemonic_progress_bar.visibility = View.GONE }
```

Comment:
why doOnEvent and not doOnTerminate?

also `.visible(false)` to hide/show view

Reply:
There is no `.doOnTerminate` for `Single` 😉